steve/job.go

```diff
@@ -0,0 +1,54 @@
+package steve
+
+import (
+	"context"
+	"io"
+	"time"
+)
+
+type Status struct {
+	ID      ID        `json:"id"`
+	Running bool      `json:"running"`
+	Started time.Time `json:"created"`
+	Stopped time.Time `json:"stopped"`
+}
+
+type Job interface {
+	// Start the job, returns an error if the job failed to start or context was cancelled
+	Start(context.Context, io.Writer) error
+
+	// Stop the job, returns an error if the context was cancelled before job was stopped
+	Stop(context.Context) error
```

Comment:
Can't we just cancel the context from `Start()` instead?

Reply:
No, this is a long running job. The context passed to `Start()` is only to cancel the `Start()` if it takes to long to "Start" the job. Once it's running you don't want the job to end if the context is cancelled.
Same purpose for the context passed to `Stop()` it's only to cancel the stop if it takes to long to "stop" the job.